commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/FileFilter.java

```diff
@@ -73,40 +140,313 @@ static public FileFilter nullFilter()
+            }
+            else if (op.getValue().equals(SQLOperator.gte))
+            {
+                if (prefixcontainsfield)
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPrefixValue(),FieldFilterRange.makeGE(sqlfilter.getPostfixValue())));
+                else
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPostfixValue(),FieldFilterRange.makeGE(sqlfilter.getPrefixValue())));
+            }
+            else if (op.getValue().equals(SQLOperator.in))
+            {
+                String list = sqlfilter.getPostfixValue().replaceAll("^\\[|\\]$", ""); //List string representation is encapsulated in square brackets
+                hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPrefixValue(),FieldFilterRange.makeIn(list.split("\\s*,\\s*"), false)));  // each entry is comma+space delimited
+            }
+            else if (op.getValue().equals(SQLOperator.isNotNull))
+            {
+                log.info("Filter 'IsNotNull' ignored -- No concept of NULL in HPCC/ECL");
+            }
+            else if (op.getValue().equals(SQLOperator.lt))
+            {
+                if (prefixcontainsfield)
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPrefixValue(),FieldFilterRange.makeLT(sqlfilter.getPostfixValue())));
+                else
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPostfixValue(),FieldFilterRange.makeLT(sqlfilter.getPrefixValue())));
+            }
+            else if (op.getValue().equals(SQLOperator.lte))
+            {
+                if (prefixcontainsfield)
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPrefixValue(),FieldFilterRange.makeLE(sqlfilter.getPostfixValue())));
+                else
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPostfixValue(),FieldFilterRange.makeLE(sqlfilter.getPrefixValue())));
+            }
+            else if (op.getValue().equals(SQLOperator.neq) || op.getValue().equals(SQLOperator.neq2))
+            {
+                if (prefixcontainsfield)
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPrefixValue(),FieldFilterRange.makeNE(sqlfilter.getPostfixValue())));
+                else
+                    hpccfilter = new FileFilter(new FieldFilter(sqlfilter.getPostfixValue(),FieldFilterRange.makeNE(sqlfilter.getPrefixValue())));
+            }
+            else
+                throw new Exception("Invalid filter operator detected: '" + op.getValue() + "' In filter: '" + sqlfilter.toString() + "'");
+        }
+        else
+            throw new Exception("Invalid filter clause: " + sqlfilter.toString());
+
+        log.debug("Converted SQL filter: '" + sqlfilter.toString() + "' to HPCC filter: '" + hpccfilter);
+        return hpccfilter;
+    }
+
+    public static void main(String [] args)
+    {
+        try
```

Comment:
Should this be extracted out into a test?

Reply:
It is, I'll remove main